File: ch04_thermalHydraulics/matlab/axial_solution_plot.m
clear
close all

analtyic_fname = '../data/axial_analytic_results.csv';
data = csvread(analtyic_fname,1,0);
LW = 2;

z    = data(:,1);
qp   = data(:,2);
h    = data(:,3);

Tinf = data(:,4);
TC = data(:,5);
TB = data(:,6);
TF = data(:,7);
T0 = data(:,8);

Tcool = data(:,9);
Tclad = data(:,10);
Tbond = data(:,11);
Tfuel = data(:,12);

tester_fname = '../data/axial_test_results.csv';
data = csvread(tester_fname,1,0);
z_tester = data(:,1);
h_tester = data(:,2);
Tfuel_tester = data(:,3);
Tbond_tester = data(:,4);
Tclad_tester = data(:,5);
Tcool_tester = data(:,6);
T0_tester = data(:,7);

% figure
% hold on
% plot(h,z,'LineWidth',LW);
% plot(h_tester,z_tester,'LineWidth',LW);
% hold off
% legend({'Analtyic','Model'},'Location','SouthEast')
% xlabel('h [kJ/kg]')
% ylabel('z [m]')
% ylim([0,z(end)])
% title('Enthalpy')

% figure
% hold on
% plot(Tinf,z);
% plot(TC,z);
% plot(TB,z);
% plot(TF,z);
% plot(T0,z);
% hold off
% legend({'T_{\infty}','T_C','T_B','T_F','T_0'},'Location','SouthEast')
% ylim([0,z(end)])
% title('Surface Temperature')

% figure
% hold on
% plot(Tcool,z);
% plot(Tclad,z);
% plot(Tbond,z);
% plot(Tfuel,z);
% hold off
% legend({'T_{cool}','T_{clad}','T_{bond}','T_{fuel}'},'Location','SouthEast')
% ylim([0,z(end)])
% title('Analytic Average Temperature')

figure
hold on
plot(Tcool_tester,z_tester,'-','LineWidth',LW);
plot(Tclad_tester,z_tester,'--','LineWidth',LW);
plot(Tbond_tester,z_tester,'-.','LineWidth',LW);
plot(Tfuel_tester,z_tester,':','LineWidth',LW);
hold off
legend({'T_{cool}','T_{clad}','T_{bond}','T_{fuel}'},'Location','SouthEast')
ylim([0,z(end)])
xlabel('Temperature [K]')
ylabel('Elevation [m]')
title('Modeled Axial Temperature')
set(gca,'FontName','Times New Roman','FontSize',12);
print('../figs/axial_temp_plot.eps','-depsc2');
% close(gcf);

% matlab_blue = [0.00 0.447 0.741];
% matlab_orange = [0.91 0.41 0.17];
% figure
% hold on
% plot(Tcool_tester,z_tester,'Color',matlab_blue);
% plot(Tcool,z,'Color',matlab_orange);
% plot(Tclad_tester,z_tester,'Color',matlab_blue);
% plot(Tbond_tester,z_tester,'Color',matlab_blue);
% plot(Tfuel_tester,z_tester,'Color',matlab_blue);
% plot(Tclad,z,'Color',matlab_orange);
% plot(Tbond,z,'Color',matlab_orange);
% plot(Tfuel,z,'Color',matlab_orange);
% hold off
% legend({'Model Average Temperature','Analytic Average Temperature'},'Location','SouthEast')
% ylim([0,z(end)])
% ylabel('z [m]')
% xlabel('Temperature [K]')
% title('Temperature Comparison')

% project analytic solution onto model grid
Tcool_project = interp1(z,Tcool,z_tester);
figure('pos',[1000 1000 850 1000])
hold on
yyaxis left
ylabel('Temperature [K]')
plot(z,Tcool,'LineWidth',LW);
plot(z_tester,Tcool_tester,'LineWidth',LW);
yyaxis right
h = plot(z_tester,Tcool_project-Tcool_tester,'-o','LineWidth',LW);
set(h,'MarkerFaceColor',get(h,'Color'));
hold off
legend({'T_{cool} Analytic','T_{cool} Model','Analytic-Model'},'Location','southoutside')
title('Temperature Difference')
xlabel('z [m]')
xlim([0,z(end)])
ylabel('Temperature Difference [K]')
set(gca,'FontName','Times New Roman','FontSize',20);
print('../figs/axial_difference_plot.eps','-depsc2');
% close(gcf);

File: ch04_thermalHydraulics/python/axial_solution_plot.py
import numpy as np
import matplotlib.pyplot as plt

LW = 2
FN = "Times New Roman"
FS = 12
plt.rc("lines", lw=LW)
plt.rc("mathtext", fontset="stix")
plt.rc("font", family=FN, size=FS)

analytic_fname = "../data/axial_analytic_results.csv"
data = np.loadtxt(analytic_fname, delimiter=",", skiprows=1)
z = data[:, 0]
qp = data[:, 1]
h = data[:, 2]
Tinf = data[:, 3]
TC = data[:, 4]
TB = data[:, 5]
TF = data[:, 6]
T0 = data[:, 7]
Tcool = data[:, 8]
Tclad = data[:, 9]
Tbond = data[:, 10]
Tfuel = data[:, 11]

tester_fname = "../data/axial_test_results.csv"
data = np.loadtxt(tester_fname, delimiter=",", skiprows=1)
z_tester = data[:, 0]
h_tester = data[:, 1]
Tfuel_tester = data[:, 2]
Tbond_tester = data[:, 3]
Tclad_tester = data[:, 4]
Tcool_tester = data[:, 5]
T0_tester = data[:, 6]

plt.figure()
plt.plot(Tcool_tester, z_tester, "-")
plt.plot(Tclad_tester, z_tester, "--")
plt.plot(Tbond_tester, z_tester, "-.")
plt.plot(Tfuel_tester, z_tester, ":")
plt.legend(["$T_{cool}$", "$T_{clad}$", "$T_{bond}$", "$T_{fuel}$"])
plt.ylim((0, z[-1]))
plt.xlabel("Temperature [K]")
plt.ylabel("Elevation [m]")
plt.title("Modeled Axial Temperature")
plt.tight_layout()
plt.savefig("../figs/axial_temp_plot.pdf")

# project analytic solution onto model grid
Tcool_project = np.interp(z_tester, z, Tcool)

plt.figure()
plt.ylabel("Temperature [K]")
plt.xlabel("z [m]")
p1 = plt.plot(z, Tcool, label="$T_{cool}$ Analytic")
c1 = p1[0]._color
p2 = plt.plot(z_tester, Tcool_tester, "--", label="$T_{cool}$ Model")
c2 = p2[0]._color
p2[0]._color = c1
ax1 = plt.gca()
ax2 = ax1.twinx()
ax2.set_ylabel("Temperature Difference [K]")
p3 = ax2.plot(
    z_tester, Tcool_project - Tcool_tester, "-o", c=c2, label="Analytic-Model"
)
p = p1 + p2 + p3
labs = [l.get_label() for l in p]
ax1.legend(p, labs, loc="lower center")
plt.title("Temperature Difference")
plt.xlim((0, z[-1]))
plt.tight_layout()
plt.savefig("../figs/axial_difference_plot.pdf")

plt.show()
